sbi/neural_nets/critic.py

```diff
@@ -119,32 +92,32 @@ def build_linear_classifier(
     check_embedding_net_device(embedding_net=embedding_net_y, datum=batch_y)
 
     # Infer the output dimensionalities of the embedding_net by making a forward pass.
-    x_numel = embedding_net_x(batch_x[:1]).numel()
-    y_numel = embedding_net_y(batch_y[:1]).numel()
+    x_numel = embedding_net_x(batch_x[:1]).size(-1)
```

Comment:
what is the reason for this change? Was this a bug?

Reply:
`.numel()` returns the number `t.flatten().shape()[0]`, but this gives only the last dimension. I like thinking about dimensions when I can. Personally I think this makes more sense since we force the user to give (Batched) 1d inputs to these networks.